core/utils/toolbox.js

```diff
@@ -71,12 +71,12 @@ Blockly.utils.toolbox.ButtonOrLabelInfo;
  * @typedef {{
  *            kind:string,
  *            name:string,
- *            id:?string,
- *            categorystyle:?string,
- *            colour:?string,
- *            cssconfig:?Blockly.ToolboxCategory.CssConfig,
- *            contents:!Array<Blockly.utils.toolbox.ToolboxItemInfo>,
- *            hidden:?string
+ *            id:(string|undefined),
+ *            categorystyle:(string|undefined),
+ *            colour:(string|undefined),
+ *            cssconfig:(!Blockly.ToolboxCategory.CssConfig|undefined),
+ *            contents:!Array<!Blockly.utils.toolbox.ToolboxItemInfo>,
```

Comment:
nit, but if the order doesn't matter here then it would be nice to have all the required ones together at the top

Reply:
done